packages/core/src/providers/studio/index.ts

```diff
@@ -120,74 +124,89 @@ export class StudioLivepeerProvider extends BaseLivepeerProvider {
     return studioStreamSessions;
   }
 
-  async createAsset(args: CreateAssetArgs): Promise<Asset> {
-    if (args.url) {
-      const createdAsset = await this._create<
-        { asset: StudioAsset },
-        Omit<CreateAssetArgs, 'file'>
-      >('/asset/upload/url', {
-        json: {
-          name: args.name,
-          url: args.url,
-        },
-        headers: this._defaultHeaders,
-      });
+  async createAsset(args: CreateAssetArgs): Promise<Asset[]> {
+    const { sources, onUploadProgress } = args;
 
-      return this.getAsset(createdAsset?.asset?.id);
-    }
+    const uploadProgress = {
+      average: 0,
+      sources: [] as CreateAssetFileProgress[],
+    };
 
-    const uploadReq = await this._create<
-      { tusEndpoint: string; asset: { id: string } },
-      Omit<CreateAssetArgs, 'file'>
-    >('/asset/request-upload', {
-      json: {
-        name: args.name,
-      },
-      headers: this._defaultHeaders,
-    });
-    const {
-      tusEndpoint,
-      asset: { id: assetId },
-    } = uploadReq;
-
-    await new Promise<void>((resolve, reject) => {
-      if (!args.file) {
-        return reject('No file provided.');
-      }
+    const assets = await Promise.allSettled(
+      sources.map(async (source, index) => {
+        const uploadReq = await this._create<
+          { tusEndpoint: string; asset: { id: string } },
+          Omit<CreateAssetSource, 'file'>
+        >('/asset/request-upload', {
+          json: {
+            name: source.name,
+            meta: source.meta,
```

Comment:
The `meta` parameter was removed in the latest changes, let's not add that back here

Reply:
See https://discord.com/channels/423160867534929930/1034826890952855593